### ISSUE TITLE
TiffImagePlugin: ValueError: need more than 0 values to unpack

### ISSUE THREAD

**Issue by edudar** [python-pillow/Pillow#1722]
An attempt to get EXIF data for auto-rotation of http://images.dpchallenge.com/images_challenge/0-999/622/1200/Copyrighted_Image_Reuse_Prohibited_457512.jpg leads to the following error:
```
  File "/opt/anaconda2/lib/python2.7/site-packages/PIL/JpegImagePlugin.py", line 391, in _getexif
    return _getexif(self)
  File "/opt/anaconda2/lib/python2.7/site-packages/PIL/JpegImagePlugin.py", line 426, in _getexif
    exif = dict(_fixup_dict(info))
  File "/opt/anaconda2/lib/python2.7/site-packages/PIL/JpegImagePlugin.py", line 407, in _fixup_dict
    return dict([(k, _fixup(v)) for k, v in src_dict.items()])
  File "/opt/anaconda2/lib/python2.7/_abcoll.py", line 414, in items
    return [(key, self[key]) for key in self]
  File "/opt/anaconda2/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 854, in __getitem__
    self._setitem(tag, handler(self, data, legacy), legacy)
  File "/opt/anaconda2/lib/python2.7/site-packages/PIL/TiffImagePlugin.py", line 534, in _setitem
    dest[tag], = values
ValueError: need more than 0 values to unpack
```

**Comment by wiredfool** [comment]
The general problem here is that the count of values in the exif don't match the Tiff metadata spec as we have it encoded, and this portion barfs if the item doesn't match. 

We've fixed portions of this before, but clearly not this part.

**Comment by mgalgs** [comment]
@edudar this seems to have been fixed by c50503a14d5b04dd42030adb77b05732e800b153 in #1988.  At least that fixed my issue that had a very similar symptom to this.  Care to check?

**Comment by homm** [comment]
This is not fixed in #1988 Rather will be fixed in #2024.

**Comment by hugovk** [comment]
PR https://github.com/python-pillow/Pillow/pull/2024 is merged, and will be released in Pillow 3.4.0 next week.
